source/utilities.cc

```diff
@@ -1363,21 +1371,26 @@ namespace aspect
                                ExcMessage (std::string("Could not open file <") + filename + ">."));
                 }
 
-
               // Read data from disk
               std::stringstream datastream;
-              filestream >> datastream.rdbuf();
 
-              if (!filestream.eof())
+              try
+                {
+                  boost::iostreams::filtering_istreambuf in;
+                  if (filename.find(".gz") == filename.size()-3)
```

Comment:
same here. or maybe introduce a local variable that stores whether you're dealing with a compressed file.